### PR TITLE
feat(analytics): add `sessionized_metrics` for disputes analytics

### LINKED ISSUE #6417: feat(analytics): add `sessionized_metrics` for refunds (backwards compatibility) and `currency_conversion`
## Current Behaviour

- There is no `sessionized_metrics` module for refunds like there is for `payments` and `payment_intents`
- Update the existing metrics with `currency_conversion` usage.
## Proposed Changes

- create `sessionized_metrics` module for refunds
- implement `currency_conversion` for refunds analytics

### LINKED ISSUE #6572: feat(analytics): add `sessionized_metrics` for disputes (backwards compatibility)
## Current Behaviour

- There is no `sessionized_metrics` module for disputes like there is for `payments` and `payment_intents`

## Proposed Changes

- create `sessionized_metrics` module for disputes

### PR DESCRIPTION
**Pull request by maverox**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Add sessionized metrics module and implement backwards compatibility.
- Add the `sessionizer_dispute` table support for Sessionized metrics.
### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

- Dispute metrics didn't have a `sessionized_metrics` module and implementation as were there in payments and payment_intents metrics

## How did you test it?

```bash
curl --location 'http://localhost:8080/analytics/v1/org/metrics/disputes' \
--header 'Accept: */*' \
--header 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--header 'Origin: http://localhost:9000' \
--header 'QueryType: SingleStat' \
--header 'Referer: http://localhost:9000/' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-site' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36' \
--header 'api-key: dev_r3kVZ4URj3fMiHhJJuhHOVqteaXwKyEMGCaFo6PiCS6S9vZsb0ErV4kCVqogM60H' \
--header 'authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiMDZlOGRmZWYtMzI0Zi00Yjc0LTg2MTItYzdkZjNhOGZmZTcyIiwibWVyY2hhbnRfaWQiOiJtZXJjaGFudF8xNzI3NDM0NTkzIiwicm9sZV9pZCI6Im9yZ19hZG1pbiIsImV4cCI6MTczMTY1MTcyMCwib3JnX2lkIjoib3JnXzl3Y0kxQ2hTOGVEdnRRQmRXak9IIiwicHJvZmlsZV9pZCI6InByb19FNW5lek43YjZUbVB1WlUzbEU1VSJ9.gtctUL339INfdj6CVmZLW2NDMNcseMTCPX0uxSsOD1c' \
--header 'sec-ch-ua: "Chromium";v="130", "Google Chrome";v="130", "Not?A_Brand";v="99"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "macOS"' \
--data '[
    {
        "timeRange": {
            "startTime": "2024-11-06T18:30:00Z",
            "endTime": "2024-11-14T10:51:00Z"
        },
        "mode": "ORDER",
        "source": "BATCH",
        "metrics": [
            "dispute_status_metric",
            "total_amount_disputed",
            "total_dispute_lost_amount",
            "sessionized_dispute_status_metric",
            "sessionized_total_amount_disputed",
            "sessionized_total_dispute_lost_amount"
        ],
        "delta": true
    }
]'
```
Screenshot 2024-11-14 at 6 14 32 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/analytics/src/disputes/accumulators.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ pub struct RateAccumulator {
 }
 #[derive(Debug, Default)]
 #[repr(transparent)]
-pub struct SumAccumulator {
+pub struct PaymentProcessedAmountAccumulator {
```
Comment (tsdk02) [inline crates/analytics/src/disputes/accumulators.rs:20]: We can name it as `DisputedAmountAccumulator`

File: crates/analytics/src/disputes/accumulators.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ pub trait DisputeMetricAccumulator {
     fn collect(self) -> Self::MetricOutput;
 }
 
-impl DisputeMetricAccumulator for SumAccumulator {
+impl DisputeMetricAccumulator for PaymentProcessedAmountAccumulator {
```
Comment (tsdk02) [inline crates/analytics/src/disputes/accumulators.rs:32]: `DisputedAmountAccumulator` here as well

Comment (maverox) [reply crates/analytics/src/disputes/accumulators.rs]: done

File: crates/analytics/src/disputes/accumulators.rs (hunk 1 above)
Comment (maverox) [reply crates/analytics/src/disputes/accumulators.rs]: done

**Review by ShivanshMathurJuspay — APPROVED** [review]

**Pull request merged**